src/components/ElementBar/ColorPickerButton/useFirstNonTransparentColor.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useMemo } from 'react';
+import { Element } from '../../../state';
+
+/**
+ * Hook that returns the first non-transparent colour of the passed elements.
+ * If no colour can be determined it falls back to the colour of the first element.
+ */
+export const useFirstNonTransparentColor = (elements: Element[]) => {
+  const color = useMemo(() => {
+    const colorElement =
+      elements.find((element) => {
+        if (element.type === 'shape' && element.fillColor !== 'transparent') {
+          return true;
+        }
+
+        if (element.type === 'path' && element.strokeColor !== 'transparent') {
+          return true;
+        }
+
+        return false;
+      }) ?? elements[0];
+
+    return colorElement?.type === 'path'
+      ? colorElement?.strokeColor
+      : colorElement?.fillColor;
+  }, [elements]);
+  return color;
+};
```

Comment:
I think that by jsdoc it is not really clear that it can return `'transparent'` color. It is not clear that it can return `undefined` as well. The signature has no return type but should be `string | undefined` here. Please use explicit types for public apis. It is a pattern across the code.

I think these `'transparent'` and `undefined` cases could be handled more explicit in implementation.

Additionally, does it needs to be a hook? It uses `useMemo` and `elements`, but `elements` will be a new instance each time because it is called as `useFirstNonTransparentColor(Object.values(activeElements))` from `ColorPicker`. We can use the `Elements` object here but it will not actually save to much rerenderings because there are just a few. In general this function will exit as soon as it finds not `'transparent'` color. So I suggest to make it a simple function.

Reply:
- Converted to a function
- Renamed to `extractFirstNonTransparentOrFirstColor`
- Updated the doc to mention the transparent and undefined cases